sql/src/main/java/io/crate/metadata/functions/Signature.java

```diff
@@ -308,6 +312,10 @@ public TypeSignature getReturnType() {
         return returnType;
     }
 
+    public boolean variableArity() {
+        return variableArity;
```

Comment:
Could this be moved to the `SignatureBindingInfo`, or do we need this always available?

Reply:
`SignatureBindingInfo` is not serializable, see https://github.com/crate/crate/blob/280cb15b80b499f466edb69ab010c89a3225fd1d/sql/src/main/java/io/crate/metadata/functions/SignatureBindingInfo.java#L29-L33. Therefore, `signature#getSignatureBindingInfo` can retrun null. 

I thought of serializing `SignatureBindingInfo`, but then it appeared that we just need `variableArity` to fill some fields of the `pg_proc`, therefore, I've serialized it with a signature object.

Reply:
> I thought of serializing SignatureBindingInfo, but then it appeared that we just need variableArity to fill some fields of the pg_proc, therefore, I've serialized it with a signature object.

`SignatureBindingInfo` was created to avoid streaming of information which is only needed for binding/matching of signatured but not needed for resolving a signature to an impl directly against the registry. 
Inside the registry, `signature#getSignatureBindingInfo` is never null.

Reply:
To clarify more, `variableArity` should never be streamed.

Reply:
I'll remove that. I thought, that it won't be always possible to use `Sigmature#getBindingInfo` to get `variableArity` in order to populate the `pg_proc` table.